commit f0d2c65f17cab8cfaf4d39f7f8e2254824cd4092
Author: Peter Eisentraut <peter@eisentraut.org>
Date:   Fri Jun 5 11:18:11 2020 +0200

    OpenSSL 3.0.0 compatibility in tests
    
    DES has been deprecated in OpenSSL 3.0.0 which makes loading keys
    encrypted with DES fail with "fetch failed".  Solve by changing the
    cipher used to aes256 which has been supported since 1.0.1 (and is
    more realistic to use anyways).
    
    Note that the minimum supported OpenSSL version is 1.0.1 as of
    7b283d0e1d1d79bf1c962d790c94d2a53f3bb38a, so this does not introduce
    any new version requirements.
    
    Author: Daniel Gustafsson <daniel@yesql.se>
    Discussion: https://www.postgresql.org/message-id/flat/FEF81714-D479-4512-839B-C769D2605F8A%40yesql.se

diff --git a/src/test/ssl/Makefile b/src/test/ssl/Makefile
index 06e37d5269..777ee39413 100644
--- a/src/test/ssl/Makefile
+++ b/src/test/ssl/Makefile
@@ -79,7 +79,7 @@ ssl/server-ss.crt: ssl/server-cn-only.key ssl/server-cn-only.crt server-cn-only.
 
 # Password-protected version of server-cn-only.key
 ssl/server-password.key: ssl/server-cn-only.key
-	openssl rsa -des -in $< -out $@ -passout 'pass:secret1'
+	openssl rsa -aes256 -in $< -out $@ -passout 'pass:secret1'
 
 # Client certificate, signed by the client CA:
 ssl/client.crt: ssl/client.key ssl/client_ca.crt
diff --git a/src/test/ssl/ssl/server-password.key b/src/test/ssl/ssl/server-password.key
index 337054fe4f..a8e383a949 100644
--- a/src/test/ssl/ssl/server-password.key
+++ b/src/test/ssl/ssl/server-password.key
@@ -1,30 +1,30 @@
 -----BEGIN RSA PRIVATE KEY-----
 Proc-Type: 4,ENCRYPTED
-DEK-Info: DES-CBC,6FCB918813C446D2
+DEK-Info: AES-256-CBC,B335CBE53A05F4FC5805FC038BA80BA0
 
-rNHoFsQpQh1UGbh2QSgtWj/pkoe29Q+uZgEwbCTIfE/ONMoiZLC6f2g+PDPg2qYn
-BC4VY2NbxkOLw7kK3k6cHwRD5YWfzm+HxmXQYsfPCeGV+C/Pse8srswh7GERSQ77
-zEn7BzkXnevL8ytYAoqezYjXaDSh+76KSN64R21zaGvgToQRGICCevbnjnNRI+HJ
-sPGGFTRIWLNNnxULHYpFbMXoJSnOrCbpzEqshUFkdGTpLvLW9se3/LVMWy/Wssje
-GwQbh9mDavUlFUl+vTn+ER4C0pzV10DjtQzkDhx88OIhdTsPQ8ClIvVFpBI6fjME
-je1n4YkLWGfJrXpifg8hAedmQ9tiw/E+h8IjZJfUgL+C2QybCQ8Yx1/v8HUYzwkA
-O7WMM90qI2sxzKWJZVvRVIY7zaTzFcvhpZrceX+d7iLV3h9UF4ru0CcUqK13t3ax
-4oOvQnj8DP0fThQOloQX4zY1oUqgLNFdoqgwMIssHzCdsyZ5VdFz8iWA1n+syzKC
-DGPJHo4DQ/YNpy0gGmBUcHaiyeKsaDYBQBwE7xm3UnrtYCr1WKkcYfncuNXCZdDJ
-X8qiB1w6diDKor9NC15uBBVjv3ERCx1rgG98+KvenQob9dtd6iguLjWtUsQoGI8D
-mgcwIuw4NRMIxZa5+1ITOkjnUfvXGTVHwvjHzz2BK6gFQu2NgwgZyrgqSMOZAn3T
-JcDDEhE8lgAsEPHVsPiLuc7DbH/ehkv8VocqJE7K9vmdR2UVG2wOO7p/1Ewm0RfJ
-9wsry4Ae0uqH+ay/TfMUBOC6tEqcPzicCMc9mYEOdnAjPwoDFFXLNdbjV8WG1KTr
-zzn9cucFvbyDOKecWbrduTbEFU5LbTddlOwWBaYVlS8xDX8wYTTyK0HnpLtP5KRJ
-69D+CYOnYAQyiPqRnZglIo4hSbNsiWThhWUlbGBWwynhR6fX4MA8PcoFnR88q3g3
-Wjmq0A0jIETwBxYrLFwvgi/PtQJ2p+yjhHCsl3idBgnGYtUiRxpCkuoCIBV1B23j
-CJB3NVoOOVEoGi9WYonLOt6vciYQhkjTkM4u3JP6rB/9kDIRpW4HeXYzrhmo0JA2
-PtRbvEwJyMHNoyNc/Pf1ydb+lqv1TBUbPQLsNbzHLKklfAMhQGd3OJWn8NcGB4i7
-uXAHSA/zY5AuE4Vf6mdjCznKbqCGurT+CUctFBajLcV2gwaFT/DikzlzR+TdFJ8E
-Qcg+5XUzg7E7tTrLGjM/DFES7ECkpr1cYGheCgoZLj0lT3bLURUUHG4e6BnyGS6M
-AdeKh4mZL3HAcbI8rSyoTx0cszQu3BR4QeAMJcnsq69uuLZ3A/r5zYDTIerz53ZO
-JYGhWWEsRszToO5eGGKpONIPTIYRLtj5NBfoQ7PZGRfRBlX44Vv3JytsBjrpABJI
-iqge5jifhTKcWVB0/OAu3A3uKXOU7AozXa79sdTVRs2KoVAgRTDK7zy3FsYRsT03
-sBdZgR1pr90PhDjlJ6N3Ebk9MXbB9bcwx0C6C/aZPFTsXT6sEoPqlrd6gCLc/YU/
-uTaXmARj45I89uRcZtroq5dZy5HE5nBpds7Vydiz1MwBnIMyBRjjAg==
+1U4+GwI8FGpcrk+9uzMlQU5UZ9xOJMOZb9xA3IYMw+2BLF7zVbAkYyyiqF2pKUmi
+doOYFOGIXNV1VhVwlw674SMN+PIg72b2F7DDrqEYlicLCU4o7eeGhoiIKzTksRTU
+YV3nYCCDZCEw7V+pFeGCUAc9fc+Y0BGMYIshLVdlTYgVjZScL5kHuD9t8xa6AaTS
+mQp3jInRnHjEJbRSZnFQ9CR1LUtmGE02TOcWzoGshFdwCdtO/lJzBmmMxoL/qV1R
+Cqc0PKSANsbgvTJMriZXYSFjpMYXmxBQXDYNuFfwq67bssAVIpTSvWu9SfcY/JwV
+OqERcb1zPgDmprDvd/L7Vh/cdEWWWewOVoUo89cT7CrLvMINHqE6smM2x1xv91BB
+AOpyoGJliPGAcLDVJINm9zC1ErEjSEcR/VumZKsgSTsBYgyYezTPQYAfe+h820rs
+eC4GMu+zr31U2TVLYcb4j2t19fTgaQBj/LH3OBse9+0quoJhzmDjKelS3O3BaF05
+DM20tJRHANM+1WQ9+aFinXa1ozcGsrLSUa99oFqL4vKgL7jd0+wmCzwxaSp3rHB3
+AFHCdUOayDAdPhnGwathhAZ0AjyEJyWnA47pEpWDr7SytpbiMwOoPcW8/oKid10e
+qBK7uGK1Zc7rtckjK3CrM1VFDbxzwGbF2aKHtFFyrJtUvJwfP0Y1V2DncOsiy5Nx
+gJ3vxfi11gxnhd9VmcoY3JVvTHOsw48xYNFrZXve/X3o9eUDqb9VRs/vV3t5w+xR
+RaUPdz9cdlp2AA4xW/IvIQ7XwuBWPaPVr/g9pUvI9iJ9Z4RdruvjqDAD+ICVx9MM
+8SuN7X3gmg4mF5FEL0ct5ZdP16U8/EYvl7Np7vN3kYqbqucwCJH15R8LckAfbzIH
+yYTXC1iik4GfyN9tTpQtZsZCvV2Uo+Fo3mxP/EzB6tNbfOi3LG/coverSwgZLQsA
+Q6+Kta4PT671xXdaGLT9tEMIai9SiW5acqcdhjYvcaP69J8ZtKpNpP6HTL7IZD8p
+SbMxE9jw+bYXILR3Ie0x98z4Z04Q28/bPbvPTbXK8nv6/YpjKgq4hrRG58psHdbX
+ggS3RNzcJJMDArBka+zvbWL4jfWZhllMyGqc7q/FuoEqC5JlMTUBpru3NTNp6ZgQ
+QXRV1Pc02ff8Dp1H8FP7B7bG3E2D9eTUqR60WvmGnuAqvXgA0+4rEaUKfxELH5qc
+dZgu/yiuMttCha835wMLnOxsOJmHILwrc6/uQWydx3vNEWFx0tbV3FzVBIvqdpME
+LA4iAAz5xqvLgA5ii23Hn18ycZGU7gTERK8RdiALRzPtBW6hPreQjiMTJnBaMhXA
+Xq9opGsNmH/rZgXuk2VZ79bbl9pKN+z9ssRGzbHCVlEckfaxlrYfANwzk8PbOrZJ
+6UW3Gf2PwRRNtiVEabf0upVng7V70KSRzjfC7KBHYwbRIL4nObgTG+vc1SjgNgrx
+Ue/e8h9qiDBmgdH0Uvqfqb19HF+QzmUNoP9TVQFj+4+DuW5zN0D8weF4TuBgyHr6
+Y+Rbmq0WJlIlc8KMwX87nACesmFNSJkI0ftSLDHrLuvXRtB8f7s2cw3hd81i+scE
 -----END RSA PRIVATE KEY-----
